### ISSUE TITLE
clang_format

### ISSUE THREAD

**Issue by yanovich** [gost-engine/engine#426]
Есть предложение добавить файл настроек для `clang-format`. Это позволит установить стандарт для форматирования. Сейчас единого подхода в коде нет, поэтому приходится код руками форматировать, от чего я отвык уже )

Я пробовал разные варианты, в итоге остановился на требованиях OpenSSL для форматирования. У них тоже нет своих настроек для `clang-format`, но есть [требования](https://www.openssl.org/policies/technical/coding-style.html).

Минусом вносимого предложения является большой первоначальный whitespace патч, но плюсом -- сокращение времени на "причёсывание" будущих патчей.

Если смысл есть, сделаю PR.

[clang-format.txt](https://github.com/gost-engine/engine/files/10143482/clang-format.txt)

**Comment by beldmit** [comment]
Да, всячески одобряю. Если ещё и тест удастся сделать, будет вообще отлично

**Comment by vt-alt** [comment]
Есть `.editorconfig`.

**Comment by yanovich** [comment]
Первоначальный вариант в #427

**Comment by yanovich** [comment]
> Да, всячески одобряю. Если ещё и тест удастся сделать, будет вообще отлично

Точно можно сделать для `clang-*` вариантов тестов.

**Comment by beldmit** [comment]
Лучше бы отдельно

**Comment by phprus** [comment]
clang-format разных версий немного по разному воспринимает правила форматирования, в связи с этим автоматическая проверка в CI может приводить к сложностям, когда clang-format у автора отформатировал код одним способом, а clang-format в CI другим.

**Comment by vt-alt** [comment]
> Да, всячески одобряю. Если ещё и тест удастся сделать, будет вообще отлично

К сожалению в результате этих изменений теряется git blame, например:

View file + Blame на один из файлов из #427: https://github.com/gost-engine/engine/blame/cf83bf3cb9108370bef19284278152e14acb115a/gost_ameth.c
у большинства строк _с кодом_: **consistent formatting**

ps. Это супер: https://github.com/gost-engine/engine/blame/cf83bf3cb9108370bef19284278152e14acb115a/test_ciphers.c#L213

**Comment by phprus** [comment]
Для git blame есть штатное решение: https://michaelheap.com/git-ignore-rev/

Гитхаб его поддерживает: https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view

**Comment by yanovich** [comment]
> pps. Может достаточно переформатировать только отступы в начале строк, но не менять внутри строк?

Таких [параметров](https://clang.llvm.org/docs/ClangFormatStyleOptions.html) я не нашёл, к сожалению.

Главный эффект в том, что если такое сделать, то дальнейшее форматирование будет одной кнопкой прямо из редактора.

> Для git blame есть штатное решение: https://michaelheap.com/git-ignore-rev/
> 
> Гитхаб его поддерживает: https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view

Бомба! Можно считать, что недостатка уже нет!

**Comment by yanovich** [comment]
> Лучше бы отдельно

Сделал в #427, сработка в #428
